src/JsonRpc/OutputHandler.cs

```diff
@@ -9,42 +9,38 @@ namespace JsonRpc
 {
     public class OutputHandler : IOutputHandler
     {
-        private readonly TimeSpan _sleepTime = TimeSpan.FromMilliseconds(50);
         private readonly TextWriter _output;
         private Thread _thread;
-        private readonly ConcurrentQueue<object> _queue;
+        private readonly BlockingCollection<object> _queue;
+        private readonly CancellationTokenSource _cancel;
 
         public OutputHandler(TextWriter output)
         {
             _output = output;
-            _queue = new ConcurrentQueue<object>();
+            _queue = new BlockingCollection<object>();
+            _cancel = new CancellationTokenSource();
             _thread = new Thread(ProcessOutputQueue) {
                 IsBackground = true
             };
         }
 
-        internal OutputHandler(TextWriter output, TimeSpan sleepTime)
-            : this(output)
-        {
-            _sleepTime = sleepTime;
-        }
-
         public void Start()
         {
             _thread.Start();
         }
 
         public void Send(object value)
         {
-            _queue.Enqueue(value);
+            _queue.Add(value);
         }
+
         private void ProcessOutputQueue()
         {
+            var token = _cancel.Token;
             while (true)
             {
                 if (_thread == null) return;
-
-                if (_queue.TryDequeue(out var value))
+                if (_queue.TryTake(out var value, -1, token))
```

Comment:
To make it easier to understand what `-1` means, we should probably change it to `Timeout.Infinite`, which is a little more transparent about what it's trying to do.